uxarray/plot/accessor.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Any, overload, Optional
+
+import functools
+
+import warnings
+
+if TYPE_CHECKING:
+    from uxarray.core.dataset import UxDataset
+    from uxarray.core.dataarray import UxDataArray
+    from uxarray.grid import Grid
+
+import uxarray.plot.grid_plot as grid_plot
+import uxarray.plot.dataarray_plot as dataarray_plot
+
+
+class GridPlotAccessor:
+    _uxgrid: Grid
+    __slots__ = ("_uxgrid",)
+
+    def __init__(self, uxgrid: Grid) -> None:
+        self._uxgrid = uxgrid
+
+    def __call__(self, **kwargs) -> Any:
+        warnings.warn("Plotting for UxDataset instances not yet supported.")
+        pass
+
+
+class UxDataArrayPlotAccessor:
+    _uxda: UxDataArray
+    __slots__ = ("_uxda",)
+
+    def __init__(self, uxda: UxDataArray) -> None:
+        self._uxda = uxda
+
+    @functools.wraps(dataarray_plot.plot)
+    def __call__(self, **kwargs) -> Any:
+        return dataarray_plot.plot(self._uxda, **kwargs)
+
+    @functools.wraps(dataarray_plot.raster)
+    def raster(self,
+               plot_height: Optional[int] = 300,
+               plot_width: Optional[int] = 600,
+               x_range: Optional[tuple] = (-180, 180),
+               y_range: Optional[tuple] = (-90, 90),
+               cmap: Optional[str] = "inferno",
+               agg: Optional[str] = "mean"):
```

Comment:
I'd suggest we use *args and **kwargs in the beginning and end of the function signature to accept changeable number/definition of arguments:

 `def raster(self, *args, ... our args ..., **kwargs):`

 This could especially be useful to support any changes of args in the upstream (e.g. `datashader.transfer_functions.shade()` in this case).

Reply:
Good idea, I'll look into this. My only concern is that for someone who isn't familiar with Datashader, it may not be clear what to pass through into the function. 

Also, the default values for our application (specifically x and y ranges) are different than those that are default in Datashadder.